lib/index.ts

```diff
@@ -19,8 +19,15 @@ type Metrics = EventEmitter<Events>;
 interface RemovableImageNode extends ImageNode {
 	removed?: true;
 	children: Record<string, RemovableImageNode>;
+	parent?: RemovableImageNode;
 }
 
+const isCandidateForRemoval = (node: RemovableImageNode): boolean => {
+	if (node.isUsedByAContainer) {
+		return false;
+	}
+	return Object.values(node.children).every((n) => n.removed);
```

Comment:
@Page- What do you think about
```ts
return !node.isUsedByAContainer && Object.values(node.children).every((n) => n.removed);
```

?

Reply:
Yeah I'll switch to that, previously I was doing the filter into an array and then a length check like in the initial `getUnusedTreeLeafs` below so it couldn't be a single line (I'd really like to deduplicate the logic but couldn't think of a nice way to do it without impacting performance.. and since the whole point of this change is to improve performance that isn't something I wanted to do)